From ff4eb85467247601fb5b7b0ebc11fdd34be35b0f Mon Sep 17 00:00:00 2001
From: xrenopnn <razenxizu@gmail.com>
Date: Thu, 15 Oct 2026 01:00:15 +0530
Subject: [PATCH 4/4] [PATCH 4/4] ims: globally set RECEIVER_EXPORTED for
 registerReceiver

E AndroidRuntime: FATAL EXCEPTION: main
E AndroidRuntime: Process: org.codeaurora.ims, PID: 3692
E AndroidRuntime: java.lang.RuntimeException: Unable to create service org.codeaurora.ims.ImsService: java.lang.SecurityException: org.codeaurora.ims: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts

Co-authored-by: Cosmin Tanislav <demonsingur@gmail.com>
Signed-off-by: xrenopnn <razenxizu@gmail.com>
Co-authored-by: SGCMarkus <markusornik@gmail.com>
---
 smali/com/qualcomm/ims/vt/LowBatteryHandler.smali  | 6 ++++--
 smali/org/codeaurora/ims/ImsCallModification.smali | 6 ++++--
 smali/org/codeaurora/ims/ImsSubController.smali    | 6 ++++--
 3 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/smali/com/qualcomm/ims/vt/LowBatteryHandler.smali b/smali/com/qualcomm/ims/vt/LowBatteryHandler.smali
index 1b5620a..4bb8114 100644
--- a/smali/com/qualcomm/ims/vt/LowBatteryHandler.smali
+++ b/smali/com/qualcomm/ims/vt/LowBatteryHandler.smali
@@ -96,7 +96,7 @@
 .end method
 
 .method private constructor <init>(Ljava/util/List;Landroid/content/Context;)V
-    .locals 1
+    .locals 2
     .annotation system Ldalvik/annotation/Signature;
         value = {
             "(",
@@ -143,7 +143,9 @@
 
     invoke-direct {p0, p1}, Landroid/content/IntentFilter;-><init>(Ljava/lang/String;)V
 
-    invoke-virtual {p2, v0, p0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v1, 0x2
+
+    invoke-virtual {p2, v0, p0, v1}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     return-void
 .end method
diff --git a/smali/org/codeaurora/ims/ImsCallModification.smali b/smali/org/codeaurora/ims/ImsCallModification.smali
index fca486c..f3731c5 100644
--- a/smali/org/codeaurora/ims/ImsCallModification.smali
+++ b/smali/org/codeaurora/ims/ImsCallModification.smali
@@ -208,7 +208,7 @@
 .end method
 
 .method public constructor <init>(Lorg/codeaurora/ims/ImsCallSessionImpl;Landroid/content/Context;Lorg/codeaurora/ims/ImsSenderRxr;Landroid/os/Looper;)V
-    .locals 3
+    .locals 4
 
     .line 136
     invoke-direct {p0}, Ljava/lang/Object;-><init>()V
@@ -342,7 +342,9 @@
     .line 149
     iget-object p2, p0, Lorg/codeaurora/ims/ImsCallModification;->mContext:Landroid/content/Context;
 
-    invoke-virtual {p2, v0, p1}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v1, 0x2
+
+    invoke-virtual {p2, v0, p1, v1}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 150
     const-string p1, "Registering car mode receiver"
diff --git a/smali/org/codeaurora/ims/ImsSubController.smali b/smali/org/codeaurora/ims/ImsSubController.smali
index e5916b3..1a5d668 100644
--- a/smali/org/codeaurora/ims/ImsSubController.smali
+++ b/smali/org/codeaurora/ims/ImsSubController.smali
@@ -304,7 +304,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;Ljava/util/List;Ljava/util/List;Landroid/os/Looper;)V
-    .locals 3
+    .locals 5
     .annotation system Ldalvik/annotation/Signature;
         value = {
             "(",
@@ -409,7 +409,9 @@
 
     invoke-direct {v1, v2}, Landroid/content/IntentFilter;-><init>(Ljava/lang/String;)V
 
-    invoke-virtual {p1, v0, v1}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v2, 0x2
+
+    invoke-virtual {p1, v0, v1, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 210
     iput-object p2, p0, Lorg/codeaurora/ims/ImsSubController;->mSenderRxrs:Ljava/util/List;
-- 
2.34.1

